src/rules/noInvalidRegexpRule.ts

```diff
@@ -26,9 +26,8 @@ class NoInvalidRegexpWalker extends Lint.RuleWalker {
         const expr = (args[0] as ts.StringLiteral).text;
         const flags = args.length > 1 && args[1].kind === ts.SyntaxKind.StringLiteral ? (args[1] as ts.StringLiteral).text : undefined;
 
-        let regex: RegExp;
         try {
-          regex = new RegExp(expr, flags);
+          (() => new RegExp(expr, flags))();
```

Comment:
I think we can simply instantiate the Regex here, no need for a IIFE.

Reply:
I agree, not really sure why I had it like that haha.

Reply:
Oh wait, I think the IIFE was to get past linter rules to do with "no unused expressions". Let my change and try it out

Reply:
I can change and suppress if needed